src/tests/build.proj

```diff
@@ -460,14 +460,15 @@
 
   <!-- This target moves app bundles to the payload directory -->
   <Target Name="AfterAppleBuild">
-    <!-- Apparently MSBuild cannot move directories and recursively copying a
-         a directory requires writing some sort of recursive traversal
-         logic yourself. -->
+    <!-- MSBuild cannot move directories or recursively copy content. -->
     <ItemGroup>
-      <RecursiveCopyHack Include="$(AppBundlePath)/**/*.*" />
+      <BundleContent Include="$(AppBundlePath)/**/*.*" />
     </ItemGroup>
     <MakeDir Directories="$(FinalPath)" />
-    <Copy SourceFiles="@(RecursiveCopyHack)" DestinationFolder="$(FinalPath)/%(RecursiveDir)" />
+    <!-- MacCatalyst requires Contents/Info.plist and Contents/MacOS/ApplicationBinary -->
+    <Copy Condition="'$(TargetOS)' == 'maccatalyst' AND '%(BundleContent.Filename)%(BundleContent.Extension)' == 'Info.plist'" SourceFiles="@(BundleContent)" DestinationFolder="$(FinalPath)/Contents/%(RecursiveDir)" />
+    <Copy Condition="'$(TargetOS)' == 'maccatalyst' AND '%(BundleContent.Filename)%(BundleContent.Extension)' != 'Info.plist'" SourceFiles="@(BundleContent)" DestinationFolder="$(FinalPath)/Contents/MacOS/%(RecursiveDir)" />
+    <Copy Condition="'$(TargetOS)' != 'maccatalyst'" SourceFiles="@(BundleContent)" DestinationFolder="$(FinalPath)/%(RecursiveDir)" />
```

Comment:
Isn't `Contents` dir already contained in the `BundleContent`? (I am not sure why we need to special-case maccatalyst)

Reply:
It should be already there, but I couldn't recursively copy content in msbuild from the build directory to the bundle. The `RecursiveDir` metadata for `BundleContent` is empty here. Please share any ideas on how to avoid "manual" copy.

It is in place to verify E2E tests, but I think we need to resolve this before merging.

Reply:
> The RecursiveDir metadata for BundleContent is empty here. 

Yes, that is what I was confused about.